sourcegraph/sourcegraph
# Code Insights: Split dashboard page into dashboards and all insights tabs

Part of https://github.com/sourcegraph/sourcegraph/issues/43561
[Designs](https://www.figma.com/file/1eai7vAekht0MT1Cl9SbJ3/Pagination---all-insights-page---must-have---FE-scale-audit---strategic-readiness-%2343561-Review?node-id=275%3A1628&t=INjPciSOMKxhcHWd-4)

## Background
In this PR we split all insights dashboards and the other dashboards into two separate high-level tabs on the code insight main page. So, this PR introduces new routes for this page
 - `/insights/about` - this is the same rote  for the getting started page tab
 - `/insights/all` - this route for the all insights dashboard (previously it was `/insights/dashboards/all`
 - `insights/dashboards/<dashboardId>` - this route covers the dashboards tab (basically all dashboards that are created by users)

## TODO 
- [x] Split the UI 
- [x] Add empty screen UI for the dashboard tab (when you have no dashboard, previously we didn't have it since we always had at least one built-in dashboard, which is all insights dashboard) cc @AlicjaSuska (when you got a minute, please let me know about your ideas on this screen)
- [x] Fix storybook and unit tests
- [x] Fix integration tests, mocks and flows

## Test plan
- Open the `/insights` route, and you should be redirected either to the getting started page `/insights/about` (if you have zero insights) or to the all insights tab `/insights/all` if you have at least one insight
- Click on the dashboard tab, and you should be redirected to the `/insights/dashboards/`. While we're fetching the dashboard list, your URL should remain `/insights/dashboards/`, but as soon as we have the dashboard list, we redirect you to the `/insights/dashboard/<first dashboard id>`. 
- If you go to the `/insights/dashboards/<dashboard id>` you should see the dashboard resolved by ID (not interactions with URL should happen) 
- If you go from the dashboard to the all insights tab or about tab and go back to the dashboard tab, the last picked dashboard should be preserved 

## Next steps (further PRs)
In the next PRs, we will add a "load more insights" pagination to the all insights page. It will be done in a separate PR in order to simplify the review and testing process.



## App preview:

- [Web](https://sg-web-vk-move-all-insights-dashboard-2.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
